x-pack/plugins/siem/public/components/formatted_date/index.tsx

```diff
@@ -26,6 +27,12 @@ export const PreferenceFormattedDate = pure<{ value: Date }>(({ value }) => (
   </KibanaConfigContext.Consumer>
 ));
 
+export const isEpochString = (possibleEpoch: string): boolean =>
+  possibleEpoch.trim() !== '' && !isNaN(+possibleEpoch);
```

Comment:
As a heads up, this will take hex values (e.g. `'0x29'`) and return true, which I suppose is valid as the hex string will be evaluated as the appropriate decimal. Interesting no less. 

Here's a test:

``` ts
    test('returns hex string number as epoch', () => {
      expect(isEpochString('0x29')).toBe(true);
    });
```

Anyway, only issue I can see here is if the ingestion date format lacks separators like  `YYYYMMMDD` (or similarly the [basic_date from ES](https://www.elastic.co/guide/en/elasticsearch/reference/6.1/mapping-date-format.html#built-in-date-formats)), in which case this would pick it up and think it's an epoch. Not sure how you get around that scenario without having the current format available as well. Either way, this is such a corner case I don't think there's any need to worry about it and this logic looks sounds. 👍

Reply:
Per conversations, I swapped this around and refactored a bit to be more tolerant where it tries other string situations first before falling back on trying to get it to work with a string that contains an Epoch.